vm/src/obj/objlist.rs

```diff
@@ -519,6 +519,17 @@ impl PyListRef {
         }
     }
 
+    fn ne(self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
```

Comment:
Because `eq` and `ne` shares lots part of the implementations, creating a functoin `inner_eq` like other classes will be great.